tools/BuildMonitor/BuildEventProcessor.cs

```diff
@@ -0,0 +1,362 @@
+        /// </example>
+        private const string ProtocolsPattern = @"(([a-zA-Z])+://(([a-zA-Z0-9_.\-/])+)+)";
+
+        /// <summary>
+        /// Match for user ID patterns, e.g. domain and username, UNC, et alia.
+        /// </summary>
+        /// <example>
+        /// matches:  "{user}", "domain\user", "user23@domain.tld", "userid:some-long-guid-like-string-of-text"
+        /// </example>
+        private const string IdentifierPattern = @"({([a-zA-Z0-9@.\-_])+})|(([a-zA-Z0-9.\-_])+\\([a-zA-Z0-9.\-_])+)|(([a-zA-Z0-9.\-_])+@([a-zA-Z0-9.\-_])+)|(user(\w){0,2}(\W){0,1}([a-zA-Z0-9.\-_])+)";
+
+        #endregion
+
+        #endregion Private Constants
+
+        #region Private Members
+
+        private DTE _dte;
+        private IVsSolution _solution;
+        private bool _microsoftInternalUser = false;
+
+        // build and project members
+        private Guid _currentBuildGuid;
+        private Guid _firstIslandwoodProjectGuid;
+        private string _solutionConfigRaw;
+        private string _solutionPlatform;
+        private string _solutionConfig;
+
+        // compiled regexes
+        private Regex _localPathsMatcher;
+        private Regex _networkPathsMatcher;
+        private Regex _protocolsMatcher;
+        private Regex _identifierMatcher;
+
+        private Stopwatch _processorStopwatch;
+
+        #endregion Private Members
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BuildMonitor"/> class.
+        /// </summary>
+        public BuildEventProcessor(DTE dte, IVsSolution solution, bool microsoftInternalUser)
+        {
+            if (dte == null)
+            {
+                throw new ArgumentNullException("dte");
+            }
+
+            if (solution == null)
+            {
+                throw new ArgumentNullException("solution");
+            }
+
+            _dte = dte;
+            _solution = solution;
+            _microsoftInternalUser = microsoftInternalUser;
+
+            string pathSeparator = Path.DirectorySeparatorChar.ToString();
+            _localPathsMatcher = new Regex(BuildEventProcessor.LocalPathsPattern.Replace("{path separator}", pathSeparator), RegexOptions.Compiled);
+            _networkPathsMatcher = new Regex(BuildEventProcessor.NetworkPathsPattern.Replace("{path separator}", pathSeparator), RegexOptions.Compiled);
+            _protocolsMatcher = new Regex(BuildEventProcessor.ProtocolsPattern, RegexOptions.Compiled);
+            _identifierMatcher = new Regex(BuildEventProcessor.IdentifierPattern, RegexOptions.Compiled);
+
+            _processorStopwatch = new Stopwatch();
+        }
+
+        #region Public Methods
+        public bool HasIslandwoodProjects()
+        {
+            foreach (Project project in _dte.Solution.Projects)
+            {
+                if (project.Kind.ToString().ToLowerInvariant().Equals(BuildEventProcessor.WinObjCProjectType))
```

Comment:
I suppose this is the new code in this vsix (or changed?).  This will require some eyes from the quality folks..